SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/AbstractLockingPolicy.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "GameFramework/Actor.h"
+
+#include "SpatialConstants.h"
+
+#include "AbstractLockingPolicy.generated.h"
+
+UCLASS(abstract)
+class SPATIALGDK_API UAbstractLockingPolicy : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual bool CanAcquireLock(const AActor* Actor) const PURE_VIRTUAL(UAbstractLockingPolicy::CanAcquireLock, return false;);
+	virtual bool IsLocked(const AActor* Actor) const PURE_VIRTUAL(UAbstractLockingPolicy::IsLocked, return false;);
+	virtual ActorLockToken AcquireLock(const AActor* Actor, FString LockName) PURE_VIRTUAL(UAbstractLockingPolicy::AcquireLock, return 0;);
+	virtual ActorLockToken AcquireLock(const AActor* Actor) PURE_VIRTUAL(UAbstractLockingPolicy::AcquireLock, return 0;);
```

Comment:
Is there any benefit of having two `AcquireLock` instead of one, with default argument `ActorLockToken AcquireLock(const AActor* Actor, FString LockName = "")`?

Reply:
If you wanted to not store a string at all vs storing an empty string. Arguably we shouldn't have this and can add it in later. Although refactoring public interfaces tends to be a pain for users who have made their own classes.

Reply:
I think a default `FString LockName = ""` still allows you to store nothing at all if needed. Having two functions is more code duplication, but makes the interface more explicit, so I don't mind leaving it as is.